src/main/java/cam72cam/mod/world/World.java

```diff
@@ -118,7 +118,7 @@ private void checkLoadedEntities() {
             for (int entityId : new ArrayList<>(this.entityByID.keySet())) {
                 if (this.internal.getEntityByID(entityId) == null) {
                     Entity entity = this.entityByID.get(entityId);
-                    if (entity != null) { // be super careful around concurrent modification (I should probably synchronize around the map)
+                    if (entity != null && !this.internal.loadedEntityList.contains(entity.internal)) {
```

Comment:
The change to line 121 is a no-go as it's adding back in the expensive contains() check for the loadedEntityList. This.

Reply:
This branch is only hit after the happy path (getEntityByID) fails.  It's only ever called for entities that are already questionable.

Reply:
The performance impact should be negligible and is required for proper functionality.